SPECS-EXTENDED/linuxptp/linuxptp.spec

```diff
@@ -1,72 +1,121 @@
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
-%global gitfullver e0580929f451e685d92cd10d80b76f39e9b09a97
-%global gitver %(c=%{gitfullver}; echo ${c:0:6})
 %global _hardened_build 1
-%global testsuite_ver a7f6e1
-%global clknetsim_ver 79ffe4
+%global testsuite_ver ff37e2
+%global clknetsim_ver 9ed48d
+%global selinuxtype targeted
+%bcond_with selinux
 
 Name:		linuxptp
-Version:	2.0
-Release:	8%{?dist}
+Version:	3.1.1
+Release:	1%{?dist}
 Summary:	PTP implementation for Linux
 
 License:	GPLv2+
 URL:		http://linuxptp.sourceforge.net/
 
-#Source0:	https://downloads.sourceforge.net/%{name}/%{name}-%{version}.tgz
-Source0:	https://github.com/richardcochran/%{name}/archive/%{gitver}/%{name}-%{gitver}.tar.gz
+Source0:	https://sourceforge.net/projects/%{name}/files/v3.1/%{name}-%{version}.tgz
 Source1:	phc2sys.service
 Source2:	ptp4l.service
 Source3:	timemaster.service
 Source4:	timemaster.conf
+Source5:	ptp4l.conf
 # external test suite
-Source10:	https://github.com/mlichvar/linuxptp-testsuite/archive/%{testsuite_ver}/linuxptp-testsuite-%{testsuite_ver}.tar.gz
+Source10:	linuxptp-testsuite-%{testsuite_ver}.tar.gz
 # simulator for test suite
-Source11:	https://github.com/mlichvar/clknetsim/archive/%{clknetsim_ver}/clknetsim-%{clknetsim_ver}.tar.gz
+Source11:	clknetsim-%{clknetsim_ver}.tar.gz
+# selinux policy
+Source20:	linuxptp.fc
+Source21:	linuxptp.if
+Source22:	linuxptp.te
 
-BuildRequires:	gcc gcc-c++ systemd
-BuildRequires:	net-snmp-devel
+# fix handling of zero-length messages
+Patch1:		linuxptp-zerolength.patch
+# revert phc2sys options needed by the older version of test suite
+Patch2:		clknetsim-phc2sys.patch
+
+# The following patch is a combination of multiple patches to enable HA in linuxptp
+# https://review.opendev.org/c/starlingx/integ/+/891638
+Patch3:         enable-ha.patch
+
+BuildRequires:	gcc gcc-c++ make systemd
 
 %{?systemd_requires}
 
+%if 0%{?with_selinux}
+Requires:	(%{name}-selinux if selinux-policy-%{selinuxtype})
+%endif
+
 %description
 This software is an implementation of the Precision Time Protocol (PTP)
 according to IEEE standard 1588 for Linux. The dual design goals are to provide
 a robust implementation of the standard and to use the most relevant and modern
 Application Programming Interfaces (API) offered by the Linux kernel.
 Supporting legacy APIs and other platforms is not a goal.
 
+%if 0%{?with_selinux}
```

Comment:
Is this a pattern to only conditionally build the selinux support? Seems to me we would want it always, but then decide during image construction whether to add it or not, rather than making it during build time. Or does it have some additional implications if enabled during build?

Reply:
This is the way the SPEC was imported from Fedora. But I think we should incorporate your idea about enabling/disabling SELinux in the OS image build process.

Reply:
The current spec for linuxptp 2.0 does not have selinux policies FYI.